source/common/ssl/certificate_validation_context_config_impl.h

```diff
@@ -8,14 +8,26 @@
 #include "envoy/ssl/certificate_validation_context_config.h"
 #include "envoy/type/matcher/v3/string.pb.h"
 
+#include "absl/status/statusor.h"
+
 namespace Envoy {
 namespace Ssl {
 
 class CertificateValidationContextConfigImpl : public CertificateValidationContextConfig {
 public:
-  CertificateValidationContextConfigImpl(
-      const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& config,
-      Api::Api& api);
+  static absl::StatusOr<std::unique_ptr<CertificateValidationContextConfigImpl>>
+  create(const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& context,
+         Api::Api& api) {
+    auto config = std::unique_ptr<CertificateValidationContextConfigImpl>(
+        new CertificateValidationContextConfigImpl(context, api));
```

Comment:
make_unique

Reply:
you can't with the private constructor.  I prefer the private constructor to make sure folks don't miss the initialize if there are any other callers.

Reply:
Oh right; that's an ugly c++ wart